addons/mail/static/src/discuss/core/common/thread_patch.js

```diff
@@ -32,6 +32,17 @@ const threadPatch = {
                 this.state.scrollTop,
             ]
         );
+        useExternalListener(window, "keydown", (ev) => {
+                if (
+                    (ev.key === "End" || ev.altKey && ev.key === "ArrowDown") &&
+                    this.props.thread?.composer?.isFocused &&
```

Comment:
There's now a `thread.isFocused` just made very recently: https://github.com/odoo/odoo/pull/204723

This covers the composer being focused, but also click on thread makes it focused without having focus in composer.